app/client/ui/selectBy.ts

```diff
@@ -141,9 +148,36 @@ function isValidLink(source: LinkNode, target: LinkNode) {
     }
   }
 
-  // The link must not create a cycle
-  if (source.ancestors.has(target.section.getRowId())) {
-    return false;
+  // The link must not create a cycle, unless it's only same-table cursor-links all the way to target
+  if (source.ancestors.includes(target.section.getRowId())) {
+
+    // cycles only allowed for cursor links
+    if (source.column || target.column || source.isSummary) {
+      return false;
+    }
+
+    // Walk backwards along the chain of ancestors
+    // - if we hit a non-cursor link before reaching target, then that would be an illegal cycle
+    // - when we hit target, we've verified that this is a legal cycle, so break
+    //   (ancestors further up the hierarchy past target don't matter, since once we set target.linkSrcSec = src.sec,
+    //    they would stop being ancestors of src)
+    // NOTE: we're guaranteed to hit target before the end of the array (because of the `if(...includes...)` above)
+    // ALSO NOTE: isAncestorSameTableCursorLink may be 1 shorter than ancestors, but it's accounted for by the above
+    let i = 0;
+    for (;;) {
```

Comment:
So, there's already a `if i == source.ancestors.length - 1 { throw Error }` in the body of the loop to keep it from going infinite. Maybe I could change it to `if i >=` to make it a little safer? But I think trying to push that condition into the while would be ugly, since the success condition is breaking out of the loop. Would have to do some sort of thing with a `successFlag = false`,   `if(successCondition) { successFlag = true; break; } ... /* after loop */ if(!successFlag) {  throw Error }`, whereas  the current code already covers it. Do you think it's acceptable to leave it as is or should I force that condition into the while?